SPECS/selinux-policy/selinux-policy.spec

```
@@ -71,6 +71,7 @@ Requires(pre):  coreutils
 Requires(pre):  policycoreutils >= %{POLICYCOREUTILSVER}
 Provides:       selinux-policy-base
 Provides:       selinux-policy-targeted
+Provides:       container-selinux
```

I'm generally against unversioned provides- as far as I can tell, it makes it impossible to use version constraints in conjunction with the alternate names. Do we have a good reason to not do the following?

```suggestion
Provides:       selinux-policy-base = %{version}-%{release}
Provides:       selinux-policy-targeted = %{version}-%{release}
Provides:       container-selinux = %{version}-%{release}
```

---

What is the type of compatibility are we trying to achieve? Versioning for Mariner RPMs or for RPMs from external sources? The versioning of our selinux-policy package is incomparable to Fedora/RHEL/CentOS packages. For example, this is selinux-policy `2.20220106-10`, but the container-selinux I have on my Fedora 36 system is `2.188.0-1`, and the selinux-policy-targeted is `36.10-1`.

---

Good question. From my point of view, these alternate names are for external compatibility, but are used in the context of pulling in our `selinux-policy` package. Since we already have version compatibility with external sources as a non-goal for the `selinux-policy` package, we don't need to strive for that with these `Provides`.

I'm just trying to avoid a situation in the future where we want to change these names to point somewhere else, but don't have any way to force the name to resolve to the newest reference. Easiest example to come up with is that we may want to split out these into separate installable subpackages, and in that case we would (a) `Obsoletes` the old versions of the names from their new packages and (b) `Requires` the newer version from dependent packages.

Given that we already publish unversioned provides in 2.0, though, I think this is more of a change to consider for 3.0. Resolving the comment.